### ISSUE TITLE
ng build -prod fails with two errors

### ISSUE THREAD

**Issue by rkralston** [angular/angular-cli#917]
> Please provide us with the following information:
> ---------------------------------------------------------------
1. OS?  Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)
os: win32 x64

2. Versions. Please run `ng --version`. If there's nothing outputted, please
   run in a Terminal: 
        node --version
   And paste the result here.
date it to a more recent version.
angular-cli: 1.0.0-beta.2-mobile.4
node: 6.1.0
os: win32 x64

3. Repro steps. Was this an app that wasn't created using the CLI? What change did you
   do on your code? etc.
I have built a small but extensive application. I am attaching some of the configuration files. Please let me know if you need more.

I ran ng init just to make sure I had everything for the CLI.
ng build -prod

4. The log given by the failure. Normally this include a stack trace and some
   more information.

I am not aware of a log file created by build. Here's the screen shot:

C:\Users\rkr0079\Source\Repos\nrmn-account>ng build -prod
(node:6264) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
/ Building(node:6264) DeprecationWarning: 'root' is deprecated, use 'global'
(node:6264) DeprecationWarning: 'GLOBAL' is deprecated, use 'global'
Built project successfully. Stored in "dist/".
File sizes:
fs.js:95
      throw err;  // Forgot a callback but don't know where? Use NODE_DEBUG=fs
      ^

Error: ENOTEMPTY: directory not empty, rmdir 'C:\Users\rkr0079\Source\Repos\nrmn-account\dist\vendor\ng2-bootstrap\components'
    at Error (native)

[nrmn-accounts.zip](https://github.com/angular/angular-cli/files/282010/nrmn-accounts.zip)

5. Mention any other details that might be useful.
> ---------------------------------------------------------------
> > Thanks! We'll be in touch soon.

**Comment by theimdal** [comment]
Did you run it as administrator? Also, it seems the build went fine, the error seems to come from the file sizes report you get after the build has finished.

**Comment by rkralston** [comment]
Yes, the cmd window is always admin.

Manually deleted the two empty directories and it worked fine.

**Comment by rkralston** [comment]
prod compiled. it will not run. should I be looking for bootstrap in that file?

**Comment by filipesilva** [comment]
Odd. Do you have any console errors when doing `ng serve -prod`?

**Comment by rkralston** [comment]
I agree. Here's my output:

C:\Users\rkr0079\Source\Repos\nrmn-account>ng serve -prod
(node:18032) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
Livereload server on http://localhost:49156
Serving on http://localhost:4200/
(node:18032) DeprecationWarning: 'root' is deprecated, use 'global'
(node:18032) DeprecationWarning: 'GLOBAL' is deprecated, use 'global'

Build successful - 13997ms.

Slowest Trees                                 | Total
----------------------------------------------+---------------------
BundlePlugin                                  | 8458ms
Funnel                                        | 1438ms
Funnel                                        | 1327ms
BroccoliTypeScriptCompiler                    | 1316ms
vendor                                        | 1198ms

Slowest Trees (cumulative)                    | Total (avg)
----------------------------------------------+---------------------
BundlePlugin (1)                              | 8458ms
Funnel (10)                                   | 2886ms (288 ms)
BroccoliTypeScriptCompiler (1)                | 1316ms
vendor (1)                                    | 1198ms

**Comment by filipesilva** [comment]
Hm... what about on the browser console, any errors?

**Comment by rkralston** [comment]
Nope. No errors anywhere.

I looked in the main.js. It is returning the bootstrap call; but it is buried way deep in the file.

I'm not sure how to go about debugging this to see where it is failing.

**Comment by rkralston** [comment]
When I prettify main.js, the following is on line 22916 when looked at in ff and a breakpoint is not hit:
`System.registerDynamic('main.js', [
'@angular/platform-browser-dynamic',
'@angular/common',
'@angular/core',
'./app',
'@angular/http',
'@angular/router',
'angular2-jwt'
], !0, function (a, b, c) {
'use strict';
var d = a('@angular/platform-browser-dynamic'),
e = a('@angular/common'),
f = a('@angular/core'),
g = a('./app'),
h = a('@angular/http'),
i = a('@angular/router'),
j = a('angular2-jwt');
return g.environment.production && f.enableProdMode(),
d.bootstrap(g.NrmnAccountAppComponent, [
  h.HTTP_PROVIDERS,
  i.ROUTER_PROVIDERS,
  j.AUTH_PROVIDERS,
  f.provide(e.LocationStrategy, {
    useClass: e.HashLocationStrategy
  })
]),
c.exports
}),`

**Comment by rkralston** [comment]
Tried moving to a fresh machine. Now prod fails with the following:

C:\Users\Keith\Source\Repos\nrmn-account>ng -v
(node:19576) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
angular-cli: 1.0.0-beta.2-mobile.4
node: 6.2.0
os: win32 x64

C:\Users\Keith\Source\Repos\nrmn-account>ng build -prod
(node:4232) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
Build failed.
The Broccoli Plugin: [BundlePlugin] failed with:
Error on fetch for main.js at file:///C:/Users/Keith/Source/Repos/nrmn-account/tmp/bundle_plugin-input_base_path-QZ2DUmcB.tmp/0/main.js
        Error: ENOENT: no such file or directory, open 'C:\Users\Keith\Source\Repos\nrmn-account\tmp\bundle_plugin-input_base_path-QZ2DUmcB.tmp\0\main.js.map'
    at Error (native)

The broccoli plugin was instantiated at:
    at BundlePlugin.Plugin (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\broccoli-plugin\index.js:10:31)
    at BundlePlugin.CachingWriter [as constructor] (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\broccoli-caching-writer\index.js:21:10)
    at BundlePlugin (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\angular-cli\lib\broccoli\angular-broccoli-bundle.js:11:5)
    at Angular2App._getBundleTree (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\angular-cli\lib\broccoli\angular2-app.js:424:22)
    at Angular2App._buildTree (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\angular-cli\lib\broccoli\angular2-app.js:159:21)
    at new Angular2App (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\angular-cli\lib\broccoli\angular2-app.js:53:23)
    at module.exports (C:\Users\Keith\Source\Repos\nrmn-account\angular-cli-build.js:6:10)
    at Class.module.exports.Task.extend.setupBroccoliBuilder (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\angular-cli\lib\models\builder.js:55:19)
    at Class.module.exports.Task.extend.init (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\angular-cli\lib\models\builder.js:89:10)
    at new Class (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\core-object\core-object.js:18:12)
    at Class.module.exports.Task.extend.run (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\angular-cli\lib\tasks\build.js:15:19)
    at C:\Users\Keith\Source\Repos\nrmn-account\node_modules\angular-cli\lib\commands\build.js:32:24
    at lib$rsvp$$internal$$tryCatch (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\rsvp\dist\rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\rsvp\dist\rsvp.js:1048:17)
    at lib$rsvp$$internal$$publish (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\rsvp\dist\rsvp.js:1019:11)
    at lib$rsvp$asap$$flush (C:\Users\Keith\Source\Repos\nrmn-account\node_modules\rsvp\dist\rsvp.js:1198:9)

The development build continues to work flawlessly.

Does my version of the CLI matter? If so, how should I upgrade?

**Comment by rkralston** [comment]
ng build -prod produces a main.js with System.registerDynamic("main.js" ... in the middle.

When I move that code to the end, the application runs.

Attaching the file.

Not exactly sure what I need to do to get this to build properly.

[main.zip](https://github.com/angular/angular-cli/files/284983/main.zip)

**Comment by rkralston** [comment]
I have tried updating the CLI with npm i angular-cli. This does not work. How should I update the CLI?

**Comment by rkralston** [comment]
I forced an upgrade. I got the error when building to prod that tmp resources did not exist.

I have rolled back to beta 2.

**Comment by kara** [comment]
This fails for me too on Mac. Same error in the tmp folder

**Comment by rkralston** [comment]
@kara : angular-cli: 1.0.0-beta.2-mobile.4 builds OK.

It has issues removing folders which it has emptied. It also will build the main.js in the wrong order once the project gets complicated. I tried Beta 5 to see if these issues were fixed.

However, both of these can be worked around by manually deleting empty folders and rebuilding; and manually editing the main.js and moving System.registerDynamic('main.js', . . . to the end of the file.

**Comment by gitdisrupt** [comment]
This may add some insight?
I too having similar problem: `ng build -prod` AND/OR `ng serve -prod` will not instantiate, `ng serve` instantiates OK, no errors in console. However...
Including `main.js`  MANUALLY inside a `<script>` at the end of the built `index.html` file fixes the production build. Please see: [https://github.com/angular/angular-cli/issues/982](url)

**Comment by filipesilva** [comment]
@rkralston @kara the broken map errors are fixed in beta.5 (not the mobile one though) but can be fixed in a project by changing `angular-cli-build.js` and `tsconfig.json` as shown in https://github.com/angular/angular-cli/pull/839.

Another user ran into a similar problem in `-prod` mode where nothing failed and yet the app did not load. It also seemed related to a bundling issue where some bundled code broke the rest, see https://github.com/angular/angular-cli/issues/951#issuecomment-222863004 for more details. In that case setting the package format in SystemJS for that specific lib fixed the issue. Can you try and tell me if that works for you?

**Comment by yusijs** [comment]
@filipesilva I'm still having the map issue in beta.5:

```
root@gaming:/home/ronnie/Desktop/dev/wih2# ng --version
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
angular-cli: 1.0.0-beta.5
node: 6.2.0
os: linux x64
root@gaming:/home/ronnie/Desktop/dev/wih2# ng build -prod
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
⠹ Building"{{content-for}}" has been deprecated and will be removed before RC.
⠸ Building(node:6718) DeprecationWarning: 'root' is deprecated, use 'global'
(node:6718) DeprecationWarning: 'GLOBAL' is deprecated, use 'global'
Build failed.
The Broccoli Plugin: [BundlePlugin] failed with:
Error on fetch for main.js at file:///home/ronnie/Desktop/dev/wih2/tmp/bundle_plugin-input_base_path-YvGycoHq.tmp/0/main.js
        Error: ENOENT: no such file or directory, open '/home/ronnie/Desktop/dev/wih2/tmp/bundle_plugin-input_base_path-YvGycoHq.tmp/0/home/ronnie/Desktop/dev/wih2/tmp/broccoli_type_script_compiler-input_base_path-2rjKQsJ6.tmp/0/main.js.map'
    at Error (native)

The broccoli plugin was instantiated at: 
    at BundlePlugin.Plugin (/home/ronnie/Desktop/dev/wih2/node_modules/broccoli-plugin/index.js:10:31)
    at BundlePlugin.CachingWriter [as constructor] (/home/ronnie/Desktop/dev/wih2/node_modules/broccoli-caching-writer/index.js:21:10)
    at BundlePlugin (/home/ronnie/Desktop/dev/wih2/node_modules/angular-cli/lib/broccoli/angular-broccoli-bundle.js:11:5)
    at Angular2App._getBundleTree (/home/ronnie/Desktop/dev/wih2/node_modules/angular-cli/lib/broccoli/angular2-app.js:421:22)
    at Angular2App._buildTree (/home/ronnie/Desktop/dev/wih2/node_modules/angular-cli/lib/broccoli/angular2-app.js:159:21)
    at new Angular2App (/home/ronnie/Desktop/dev/wih2/node_modules/angular-cli/lib/broccoli/angular2-app.js:53:23)
    at module.exports (/home/ronnie/Desktop/dev/wih2/angular-cli-build.js:6:10)
    at Class.module.exports.Task.extend.setupBroccoliBuilder (/home/ronnie/Desktop/dev/wih2/node_modules/angular-cli/lib/models/builder.js:55:19)
    at Class.module.exports.Task.extend.init (/home/ronnie/Desktop/dev/wih2/node_modules/angular-cli/lib/models/builder.js:89:10)
    at new Class (/home/ronnie/Desktop/dev/wih2/node_modules/core-object/core-object.js:18:12)
    at Class.module.exports.Task.extend.run (/home/ronnie/Desktop/dev/wih2/node_modules/angular-cli/lib/tasks/build.js:15:19)
    at /home/ronnie/Desktop/dev/wih2/node_modules/angular-cli/lib/commands/build.js:32:24
    at lib$rsvp$$internal$$tryCatch (/home/ronnie/Desktop/dev/wih2/node_modules/rsvp/dist/rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (/home/ronnie/Desktop/dev/wih2/node_modules/rsvp/dist/rsvp.js:1048:17)
    at /home/ronnie/Desktop/dev/wih2/node_modules/rsvp/dist/rsvp.js:331:11
    at lib$rsvp$asap$$flush (/home/ronnie/Desktop/dev/wih2/node_modules/rsvp/dist/rsvp.js:1198:9)
```

Debian Jessie.

**Comment by filipesilva** [comment]
@yusijs was that project generated with beta.5? It seems to have the mapping issue of a beta.4 project generation.

Can you try changing `angular-cli-build.js` and `tsconfig.json` as shown in #839 and tell me if it works?

**Comment by yusijs** [comment]
That sorted the build, but serve still borked (only shows the "Loading...") with no errors.

```
ronnie@gaming:~/Desktop/dev/wih2$ sudo ng serve -prod                                                                                                                                                                                                                          
(node:2156) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.                                                                                                                      
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
Livereload server on http://localhost:49152
Serving on http://localhost:4200/
"{{content-for}}" has been deprecated and will be removed before RC.                                                                                                                                                                                                           
(node:2156) DeprecationWarning: 'root' is deprecated, use 'global'                                                                                                                                                                                                             
(node:2156) DeprecationWarning: 'GLOBAL' is deprecated, use 'global'                                                                                                                                                                                                           

Build successful - 9708ms.
                                                                                                                                                                                                                                                                               
Slowest Trees                                 | Total                                                                                                                                                                                                                          
----------------------------------------------+---------------------                                                                                                                                                                                                           
BundlePlugin                                  | 8109ms                                                                                                                                                                                                                         
BroccoliTypeScriptCompiler                    | 954ms                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                               
Slowest Trees (cumulative)                    | Total (avg)                                                                                                                                                                                                                    
----------------------------------------------+---------------------                                                                                                                                                                                                           
BundlePlugin (1)                              | 8109ms                                                                                                                                                                                                                         
BroccoliTypeScriptCompiler (1)                | 954ms                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                               
                                                                                                                         
```

**Comment by rkralston** [comment]
Still fails for me. I have made the requested changes to angular-cli-build.js and tsconfig.json.

C:\Users\rkr0079\Source\Repos\nrmn-account>ng build -prod
(node:9800) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
/ Building(node:9800) DeprecationWarning: 'root' is deprecated, use 'global'
(node:9800) DeprecationWarning: 'GLOBAL' is deprecated, use 'global'
Build failed.
The Broccoli Plugin: [BundlePlugin] failed with:
Error on fetch for vendor/angular2-jwt/angular2-jwt.js at file:///C:/Users/rkr0079/Source/Repos/nrmn-account/tmp/bundle_plugin-input_base_path-VwnZ1R56.tmp/0/vendor/angular2-jwt/angular2-jwt.js
        Loading main.js
        Error: ENOENT: no such file or directory, open 'C:\Users\rkr0079\Source\Repos\nrmn-account\tmp\bundle_plugin-input_base_path-VwnZ1R56.tmp\0\vendor\angular2-jwt\angular2-jwt.js.map'
    at Error (native)

The broccoli plugin was instantiated at:
    at BundlePlugin.Plugin (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\broccoli-plugin\index.js:10:31)
    at BundlePlugin.CachingWriter [as constructor] (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\broccoli-caching-writer\index.js:21:10)
    at BundlePlugin (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\angular-cli\lib\broccoli\angular-broccoli-bundle.js:11:5)
    at Angular2App._getBundleTree (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\angular-cli\lib\broccoli\angular2-app.js:421:22)
    at Angular2App._buildTree (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\angular-cli\lib\broccoli\angular2-app.js:159:21)
    at new Angular2App (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\angular-cli\lib\broccoli\angular2-app.js:53:23)
    at module.exports (C:\Users\rkr0079\Source\Repos\nrmn-account\angular-cli-build.js:6:10)
    at Class.module.exports.Task.extend.setupBroccoliBuilder (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\angular-cli\lib\models\builder.js:55:19)
    at Class.module.exports.Task.extend.init (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\angular-cli\lib\models\builder.js:89:10)
    at new Class (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\core-object\core-object.js:18:12)
    at Class.module.exports.Task.extend.run (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\angular-cli\lib\tasks\build.js:15:19)
    at C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\angular-cli\lib\commands\build.js:32:24
    at lib$rsvp$$internal$$tryCatch (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\rsvp\dist\rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\rsvp\dist\rsvp.js:1048:17)
    at lib$rsvp$$internal$$publish (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\rsvp\dist\rsvp.js:1019:11)
    at lib$rsvp$asap$$flush (C:\Users\rkr0079\Source\Repos\nrmn-account\node_modules\rsvp\dist\rsvp.js:1198:9)



C:\Users\rkr0079\Source\Repos\nrmn-account>ng -v
(node:9232) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
angular-cli: 1.0.0-beta.5
node: 6.1.0
os: win32 x64

**Comment by rkralston** [comment]
Fixed by making the same change to file extensions for angular2-jwt : `angular2-jwt/angular2-jwt.+(js|js.map)`

I'm still having trouble with `rmdir` not deleting directories with sub-folders. I manually delete these and the build works. Perhaps `rmdir xxxx /s` might help?

**Comment by rkralston** [comment]
The prod build builds. However, `System.registerDynamic("main.js",["@angular/platform-browser-dynamic","@angular/common","@angular/core","./app","@angular/http","@angular/router","angular2-jwt"],!0,function(a,b,c){"use strict";var d=a("@angular/platform-browser-dynamic"),e=a("@angular/common"),f=a("@angular/core"),g=a("./app"),h=a("@angular/http"),i=a("@angular/router"),j=a("angular2-jwt");return g.environment.production&&f.enableProdMode(),d.bootstrap(g.NrmnAccountAppComponent,[h.HTTP_PROVIDERS,i.ROUTER_PROVIDERS,j.AUTH_PROVIDERS,f.provide(e.LocationStrategy,{useClass:e.HashLocationStrategy})]),c.exports})` get stuck in the middle of main.js and the application does not load.

This and the directories not deleting were my problems with the older build. I still have to manually intervene to get a built application.

What do I need to do to get that the main.js to be inserted at the end of the file?

**Comment by filipesilva** [comment]
@rkralston glad that it's working now, although I can't help much with the `rmdir` :/

@yusijs @rkralston as far as I can tell, the blank `loading...` screen has to do with SystemJS bundling packages without package configuration in a weird way. See https://github.com/angular/angular-cli/issues/951.

**Comment by yusijs** [comment]
@filipesilva Could you let us know the issue# @ systemjs when you've got one written down? Thanks.

**Comment by rkralston** [comment]
@filipesilva unfortunately, this is not closed. Beta 5 builds, but the build is still broken. I tried the explicit format for moment, it does not change anything. 

`System.registerDynamic("main.js",["@angular/platform-browser-dynamic","@angular/common","@angular/core","./app","@angular/http","@angular/router","angular2-jwt"],!0,function(a,b,c){"use strict";var d=a("@angular/platform-browser-dynamic"),e=a("@angular/common"),f=a("@angular/core"),g=a("./app"),h=a("@angular/http"),i=a("@angular/router"),j=a("angular2-jwt");return g.environment.production&&f.enableProdMode(),d.bootstrap(g.NrmnAccountAppComponent,[h.HTTP_PROVIDERS,i.ROUTER_PROVIDERS,j.AUTH_PROVIDERS,f.provide(e.LocationStrategy,{useClass:e.HashLocationStrategy})]),c.exports})`

still gets thrown into the middle of main.js and prevents the application from bootstrapping. When I manually move this to the end of the file, it works.

**Comment by yusijs** [comment]
@rkralston I found the easiest way to work around this for now, is to include main.js as a script tag in index.html.

**Comment by rkralston** [comment]
@yusijs thanks, that work around works for me too.

**Comment by filipesilva** [comment]
@yusijs I put up the issue number here https://github.com/angular/angular-cli/issues/951#issuecomment-223464443

@rkralston the original issue `Error: ENOTEMPTY: directory not empty,` was fixed, and then the subsequent issue is a duplicate of https://github.com/angular/angular-cli/issues/951. I understand that you still have an issue but I can't keep two open at the same time for the same thing. I will bring up on the other one that you still have a problem even with the correct format.

**Comment by filipesilva** [comment]
@rkralston actually... do you have more 3rd party libs other than moment? Because you might need to add in the format for all of them, not just moment.

**Comment by rkralston** [comment]
Yes I do. I'll post it when I get back home. Not sure about the format thing. Can you point me to documentation?





On Fri, Jun 3, 2016 at 7:50 AM -0700, "Filipe Silva" <notifications@github.com> wrote:





@rkralston actually... do you have more 3rd party libs other than moment? Because you might need to add in the format for all of them, not just moment.

---

**Comment by rkralston** [comment]
Thnx





On Fri, Jun 3, 2016 at 7:46 AM -0700, "Filipe Silva" <notifications@github.com> wrote:





@yusijs I put up the issue number here https://github.com/angular/angular-cli/issues/951#issuecomment-223464443

@rkralston the original issue `Error: ENOTEMPTY: directory not empty,` was fixed, and then the subsequent issue is a duplicate of https://github.com/angular/angular-cli/issues/951. I understand that you still have an issue but I can't keep two open at the same time for the same thing. I will bring up on the other one that you still have a problem even with the correct format.

---

**Comment by filipesilva** [comment]
The list of module formats that SystemJS supports are here: https://github.com/systemjs/systemjs/blob/master/docs/module-formats.md

Maybe start with adding the equivalent of

```
  'moment':{
    format: 'cjs'
  }
```

For all the other packages, and see if it works then.

**Comment by rkralston** [comment]
No joy. I'v attached a zip of my config files and the subsequent main.js. @yusijs 's work around still holds. You can see that main.js is being loaded in the middle of the file.
[accounts-config.zip](https://github.com/angular/angular-cli/files/303055/accounts-config.zip)

**Comment by filipesilva** [comment]
@rkralston yeap, definitely `main.js` in the middle of the file, like in https://github.com/angular/angular-cli/issues/951. 

Your configs and everything seem ok, but without checking a running project I can't really reproduce it. If you can post a repository I can debug in https://github.com/angular/angular-cli/issues/951, I can look at it and see what I can find.

**Comment by filipesilva** [comment]
@rkralston can you try with beta.6? We started bundling stuff differently and the issue should be completely fixed.

**Comment by rkralston** [comment]
Removed all instances of cli and only installed globally. When I tried the build, it complained `Cannot find module 'angular-cli/lib/broccoli/angular2-app'`.

I installed the cli locally and the build ran.

I'm still having issues with the removal of empty directories. This represents a difference in the dev and prod builds. If something in dev is moved to main.js in prod, it is not cleaning up those directories appropriately. I have to run the prod build three times to get a successful build.

SUCCESS!! main.js is now built in the proper order!!!!
